### PR TITLE
Fix "No results" banner showing up on every search

### PR DESCRIPTION
**Pull request by attfarhan**
Fixes https://github.com/sourcegraph/sourcegraph/issues/7060

Also updates all callsites of the `resultCount` field on search to use `matchCount` as `resultCount` is apparently deprecated: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@531933ca9008089f37bc0af504672718998c3288/-/blob/cmd/frontend/graphqlbackend/schema.graphql#L1378-1379

### CODE REVIEW

**Comment by attfarhan** [comment]
@lguychard yeah, it's a regression that came out of https://github.com/sourcegraph/sourcegraph/pull/7020

File: web/src/search/results/SearchResultsList.tsx [hunk 1]
```diff
@@ -398,64 +398,66 @@ export class SearchResultsList extends React.PureComponent<SearchResultsListProp
...
-                                                        : []
-                                                )}
-                                        </div>
-                                    ) : (
-                                        <>
-                                            <div className="alert alert-info d-flex m-2">
-                                                <h3 className="m-0">
-                                                    <SearchIcon className="icon-inline" /> No results
+                                    {/* Server-provided help message when there are no results*/}
+                                    {results.matchCount === 0 &&
```
Comment (felixfbecker) [inline web/src/search/results/SearchResultsList.tsx]: So the change here is that the alert is only shown when the matchCount is 0. But what if the server returns an alert that doesn't have anything to do with the result count (which is valid)? And why does the server return a "no results" alert even though there are results? Seems like this is a backend bug and this is the wrong fix?

Comment (attfarhan) [reply web/src/search/results/SearchResultsList.tsx]: > And why does the server return a "no results" alert even though there are results? 
When does this happen? The no results error doesn't get returned by the server, but we know that when `matchCount` is 0 that there are no results.

Comment (attfarhan) [reply web/src/search/results/SearchResultsList.tsx]: >But what if the server returns an alert that doesn't have anything to do with the result count (which is valid)?

Good point

**Comment by attfarhan** [comment]
(To address Felix's comment I moved the "no results" alert out of the ternary that checks for `results.alert`, so it only shows when there are no results and no alerts.

**Comment by slimsag** [comment]
Thank you so much for fixing this @attfarhan -- apologies I failed to catch this. I've filed https://github.com/sourcegraph/sourcegraph/issues/7102 for fixing the fact that automated testing did not catch this.

**Pull request merged**